android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMPBrowserActivity.java

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (C) 2020 SIL International. All rights reserved.
+ */
+
+package com.tavultesoft.kmea;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.Color;
+import android.graphics.PorterDuff.Mode;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.ImageButton;
+import android.webkit.WebChromeClient;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+import androidx.appcompat.app.AppCompatActivity;
+
+public class KMPBrowserActivity extends AppCompatActivity {
+
+  private WebView webView;
+  private static final String KMP_SEARCH_BASE_URL = "https://keyman.com/keyboards";
+  private static final String KMP_LANGUAGE = "/languages/";
+  private static final String KMP_QUERY = "?embed=linux&version="; // TODO: Update to Android
+  private boolean isLoading = false;
+  private boolean didFinishLoading = false;
+
+  @SuppressLint({"SetJavaScriptEnabled", "InflateParams"})
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    final Context context = this;
+
+    setContentView(R.layout.activity_kmp_browser);
+
+    webView = (WebView) findViewById(R.id.kmpBrowserWebView);
+    webView.getSettings().setLayoutAlgorithm(WebSettings.LayoutAlgorithm.NORMAL);
+    webView.getSettings().setJavaScriptEnabled(true);
+    webView.getSettings().setUseWideViewPort(true);
+    webView.getSettings().setLoadWithOverviewMode(true);
+    webView.getSettings().setBuiltInZoomControls(true);
+    webView.getSettings().setSupportZoom(true);
+    webView.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
+
+    webView.setWebChromeClient(new WebChromeClient() {
+      public void onProgressChanged(WebView view, int progress) {
+      }
+    });
+    webView.setWebViewClient(new WebViewClient() {
+      @Override
+      public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
+        didFinishLoading = true;
+        isLoading = false;
+      }
+
+      @Override
+      public boolean shouldOverrideUrlLoading(WebView view, String url) {
+        if (!url.toLowerCase().equals("about:blank")) {
+          if (url.startsWith("keyman:download")) {
+            // KMAPro main activity will handle this intent
+            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
+            startActivityForResult(intent, 1);
+          } else {
+            view.loadUrl(url);
```

Comment:
You should return `false` here rather than calling `view.loadUrl` to allow the default action to continue -- this handles subframes etc. A slightly cleaner pattern?